src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -417,98 +417,103 @@ void Simulator::handle_message(mavlink_message_t *msg, bool publish)
-			hil_gpos.alt = hil_state.alt / 1E3;//1E3
+		// always publish ground truth attitude message
+		int hilstate_multi;
+		orb_publish_auto(ORB_ID(vehicle_attitude_groundtruth), &_attitude_pub, &hil_attitude, &hilstate_multi, ORB_PRIO_HIGH);
+	}
 
-			hil_gpos.vel_n = hil_state.vx / 100.0f;
-			hil_gpos.vel_e = hil_state.vy / 100.0f;
-			hil_gpos.vel_d = hil_state.vz / 100.0f;
+	/* global position */
+	struct vehicle_global_position_s hil_gpos = {};
+	{
+		hil_gpos.timestamp = timestamp;
 
-			// always publish ground truth attitude message
-			int hil_gpos_multi;
-			orb_publish_auto(ORB_ID(vehicle_global_position_groundtruth), &_gpos_pub, &hil_gpos, &hil_gpos_multi,
-					 ORB_PRIO_HIGH);
-		}
+		hil_gpos.lat = hil_state.lat / 1E7;//1E7
+		hil_gpos.lon = hil_state.lon / 1E7;//1E7
+		hil_gpos.alt = hil_state.alt / 1E3;//1E3
 
-		/* local position */
-		struct vehicle_local_position_s hil_lpos = {};
-		{
-			hil_lpos.timestamp = timestamp;
-
-			double lat = hil_state.lat * 1e-7;
-			double lon = hil_state.lon * 1e-7;
-
-			if (!_hil_local_proj_inited) {
-				_hil_local_proj_inited = true;
-				map_projection_init(&_hil_local_proj_ref, lat, lon);
-				_hil_ref_timestamp = timestamp;
-				_hil_ref_lat = lat;
-				_hil_ref_lon = lon;
-				_hil_ref_alt = hil_state.alt / 1000.0f;
-			}
+		hil_gpos.vel_n = hil_state.vx / 100.0f;
+		hil_gpos.vel_e = hil_state.vy / 100.0f;
+		hil_gpos.vel_d = hil_state.vz / 100.0f;
 
-			float x;
-			float y;
-			map_projection_project(&_hil_local_proj_ref, lat, lon, &x, &y);
-			hil_lpos.timestamp = timestamp;
-			hil_lpos.xy_valid = true;
-			hil_lpos.z_valid = true;
-			hil_lpos.v_xy_valid = true;
-			hil_lpos.v_z_valid = true;
-			hil_lpos.x = x;
-			hil_lpos.y = y;
-			hil_lpos.z = _hil_ref_alt - hil_state.alt / 1000.0f;
-			hil_lpos.vx = hil_state.vx / 100.0f;
-			hil_lpos.vy = hil_state.vy / 100.0f;
-			hil_lpos.vz = hil_state.vz / 100.0f;
-			matrix::Eulerf euler = matrix::Quatf(hil_attitude.q);
-			hil_lpos.yaw = euler.psi();
-			hil_lpos.xy_global = true;
-			hil_lpos.z_global = true;
-			hil_lpos.ref_lat = _hil_ref_lat;
-			hil_lpos.ref_lon = _hil_ref_lon;
-			hil_lpos.ref_alt = _hil_ref_alt;
-			hil_lpos.ref_timestamp = _hil_ref_timestamp;
-			hil_lpos.vxy_max = std::numeric_limits<float>::infinity();
-			hil_lpos.vz_max = std::numeric_limits<float>::infinity();
-			hil_lpos.hagl_min = std::numeric_limits<float>::infinity();
-			hil_lpos.hagl_max = std::numeric_limits<float>::infinity();
-
-			// always publish ground truth attitude message
-			int hil_lpos_multi;
-			orb_publish_auto(ORB_ID(vehicle_local_position_groundtruth), &_lpos_pub, &hil_lpos, &hil_lpos_multi,
-					 ORB_PRIO_HIGH);
+		// always publish ground truth attitude message
+		int hil_gpos_multi;
+		orb_publish_auto(ORB_ID(vehicle_global_position_groundtruth), &_gpos_pub, &hil_gpos, &hil_gpos_multi,
+				 ORB_PRIO_HIGH);
+	}
+
+	/* local position */
+	struct vehicle_local_position_s hil_lpos = {};
+	{
```

Comment:
I don't see the need for these brackets?

Reply:
Must be a copy/paste error, and it looks like there was one with the `endif` you spotted as well... with your other comments I'll have to rewrite all of the commits in this PR, so you'll have to start your review over.  Sorry about that.

Reply:
I know, it's unrelated. Just thought we might as well remove them in one go.